read_replicate/common.sh

```
@@ -0,0 +1,345 @@
+    SOURCE_HOST="db.${PROJECT_ID}.supabase.co"
+    SOURCE_PORT="5432"
+    SOURCE_USER="postgres"
+    echo "==> Converted Supabase pooler URL to direct connection: $SOURCE_HOST:$SOURCE_PORT"
+  elif [[ "$SOURCE_PORT" == "6543" ]]; then
+    SOURCE_PORT="5432"
+    echo "==> Changed Supabase source port from 6543 to 5432 for direct replication"
+  fi
+
+  SOURCE_SSLMODE='require'
+  SOURCE_DB_URL="postgresql://${SOURCE_USER}:${SOURCE_PASSWORD}@${SOURCE_HOST}:${SOURCE_PORT}/${SOURCE_DB}?sslmode=${SOURCE_SSLMODE}"
+  SOURCE_CONNECTION_STRING="host=${SOURCE_HOST}
```

Shellcheck (0.11.0)

[warning] SOURCE_CONNECTION_STRING appears unused. Verify use (or export if used externally).

(SC2034)